pvlib/spectrum/mismatch.py

```diff
@@ -235,3 +236,159 @@ def integrate(e):
         smm = pd.Series(smm, index=e_sun.index)
 
     return smm
+
+
+def martin_ruiz_spectral_modifier(clearness_index, airmass_absolute,
+                                  cell_type=None, model_parameters=None):
+    r"""
+    Calculate spectral mismatch modifiers for POA direct, sky diffuse and
+    ground diffuse irradiances using the clearness index and the absolute
+    airmass.
+
+    .. warning::
+        Included model parameters for ``monosi``, ``polysi`` and ``asi`` were
+        estimated using the airmass model ``kasten1966`` [1]_. It is heavily
+        recommended to use the same model in order to not introduce errors.
+        See :py:func:`~pvlib.atmosphere.get_relative_airmass`.
+
+    Parameters
+    ----------
+    clearness_index : numeric
+        Clearness index of the sky.
+
+    airmass_absolute : numeric
+        Absolute airmass. Give attention to algorithm used (``kasten1966`` is
+        recommended for default parameters of ``monosi``, ``polysi`` and
+        ``asi``, see reference [1]_).
+
+    cell_type : string, optional
+        Specifies material of the cell in order to infer model parameters.
+        Allowed types are ``monosi``, ``polysi`` and ``asi``, either lower or
+        upper case. If not specified, ``model_parameters`` must be provided.
+
+    model_parameters : dict-like, optional
+        In case you computed the model parameters. In case any component is not
+        specified, result will have a ``np.nan`` value in its
+        corresponding value.
+        Provide either a dict or a ``pd.DataFrame`` as follows:
+
+        .. code-block:: python
+
+            # Using a dict
+            # Return keys are the same as specifying 'cell_type'
+            model_parameters = {
+                'poa_direct': {'c': c1, 'a': a1, 'b': b1},
+                'poa_sky_diffuse': {'c': c2, 'a': a2, 'b': b2},
+                'poa_ground_diffuse': {'c': c3, 'a': a3, 'b': b3}
+            }
+            # Using a pd.DataFrame
+            model_parameters = pd.DataFrame({
+                'poa_direct': [c1, a1, b1],
+                'poa_sky_diffuse':  [c2, a2, b2],
+                'poa_ground_diffuse': [c3, a3, b3]},
+                index=('c', 'a', 'b'))
+
+        ``c``, ``a`` and ``b`` must be scalar.
+
```

Comment:
```suggestion
```suggestion
        If parameters for an irradiance component (`'poa_direct'`,
        `'poa_sky_diffuse'`, or `'poa_ground_diffuse'`) are not
        specified, ``np.nan`` will be returned in the corresponding value.
```

Reply:
I don't understand what is this suggestion. I can't render the html output well with that.

Reply:
I modified the text "In case you computed..." and suggest it is moved to below the code block. Looks like the comment formatting is not correct.

Reply:
Ok, I will do something about that.

Reply:
Thank you very much @cwhanse 
I've updated the docstring, let me know what do you think of it. TIA.